Test/allocated1/dafny0/DiscoverBounds.dfy.expect

```diff
@@ -1,8 +1,8 @@
-DiscoverBounds.dfy(32,7): Warning: /!\ No terms found to trigger on.
-DiscoverBounds.dfy(33,7): Warning: /!\ No terms found to trigger on.
-DiscoverBounds.dfy(34,7): Warning: /!\ No terms found to trigger on.
-DiscoverBounds.dfy(35,7): Warning: /!\ No terms found to trigger on.
-DiscoverBounds.dfy(36,7): Warning: /!\ No terms found to trigger on.
+DiscoverBounds.dfy(32,7): Warning: /!\ No terms found to trigger on. [Related location] (DiscoverBounds.dfy(32,33)).
```

Comment:
Is the new information in the error message useful? Is it nice to get a related location without having a message explaining how the location is related?

I look at DiscoverBounds.dfy and it wasn't obvious to me how the related location is related. Could you tell me how it is?

Reply:
In general, It is always better to have a related location without having a message explaining how the location is related than having no related location at all, if the related location is important.
The information was already there, stored in nested tokens, this PR only reveals it. But I agree with your comment, it would be better if it was a nicer message here.

In `DiscoverBounds.dfy` the related location is actually the position of the trigger that was tried.

I went ahead and added information that completely replaces `[Related location]` in context, explaining how the related location is actually related. For example in the test `suppressing-warnings-behaves-properly.dfy`, before, it was saying

```
suppressing-warnings-behaves-properly.dfy(10,9): Warning: /!\ No terms found to trigger on.  [Related location] (suppressing-warnings-behaves-properly.dfy(10,33)).
suppressing-warnings-behaves-properly.dfy(11,9): Info: (Suppressed warning) No terms found to trigger on. [Related location] (suppressing-warnings-behaves-properly.dfy(11,43)).
suppressing-warnings-behaves-properly.dfy(12,9): Info: Not generating triggers for "n >= 0 || n < 0". Note that {:autotriggers false} can cause instabilities. Consider using {:nowarn}, {:matchingloop} (not great either), or a manual trigger instead. [Related location] (suppressing-warnings-behaves-properly.dfy(12,55)).
suppressing-warnings-behaves-properly.dfy(14,9): Info: Selected triggers: {f(n)} [Related location] (suppressing-warnings-behaves-properly.dfy(14,46)).
```

Now it says:


```
suppressing-warnings-behaves-properly.dfy(10,9): Warning: /!\ No terms found to trigger on. Potential trigger not suitable: (suppressing-warnings-behaves-properly.dfy(10,33)).
suppressing-warnings-behaves-properly.dfy(11,9): Info: (Suppressed warning) No terms found to trigger on. Potential trigger not suitable: (suppressing-warnings-behaves-properly.dfy(11,43)).
suppressing-warnings-behaves-properly.dfy(12,9): Info: Not generating triggers for "n >= 0 || n < 0". Note that {:autotriggers false} can cause instabilities. Consider using {:nowarn}, {:matchingloop} (not great either), or a manual trigger instead. Trigger position: (suppressing-warnings-behaves-properly.dfy(12,55)).
suppressing-warnings-behaves-properly.dfy(14,9): Info: Selected triggers: {f(n)} Trigger position: (suppressing-warnings-behaves-properly.dfy(14,46)).
```

I think it's better too! Thanks for suggesting!